### PR TITLE
fix: update newly created queries body with correct widget bindings

### LINKED ISSUE #34237: Bug: Widget bindings are not getting updated in Queries.
When we drag and drop multiple building blocks: widget bindgings are not getting updated int the queries.
For drag and drop of building blocks, Queries and jsObjects are getting refactored in the backend while widgetNames are getting refactored in the frontend.

### Approach #1 @jacquesikot 
1. send newly added actions from the backend.
2. update the bindings when widgets are being added to the canvas.
3. Run a PUT actions call to update the backend.
4. Run the added action.

### Approach #2 @AnaghHegde 
1. update the widget bindings in the backend itself. This takes care of updating the queries with correct widget names
2. on the UI  only update the widget names if there is a conflict.

> This is susceptible to performance issue regression. We have to take good care of not falling into this trap

### PR DESCRIPTION
**Pull request by jacquesikot**
## Description
When multiple blocks of the same type are dropped unto the canvas right after each other, the query body of the newly created query does not update the binding to the latest version of the widgets.

## Solution
We have gotten the newlyUpdatedQueries from the block API, then we update the actionConfiguration.body and the jsonPathKeys to match the updated binding name for the block and implement the action in the local state.

Fixes #34237

## Automation
/ok-to-test tags="@tag.Widget"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/9644500854>
> Commit: 562ad23b485afbd0c7b695b0c1aa8e8cc01c94cb
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=9644500854&attempt=1" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.Widget`

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [x] No

## Summary by CodeRabbit

- **New Features**
  - Enhanced functionality for saving building block widgets to the store.

- **Tests**
  - Updated and added new test cases for pasting building block widgets.

- **Bug Fixes**
  - Fixed issues with the import and usage of action types and selectors in the building block sagas.

### CODE REVIEW

File: app/client/src/sagas/BuildingBlockSagas/index.ts [hunk 1]
```diff
@@ -44,3 +48,71 @@ export function* saveBuildingBlockWidgetsToStore(
...
+      query.actionConfiguration.body =
+        query.actionConfiguration.body.replaceAll(oldWidgetName, newWidgetName);
+      query.jsonPathKeys = query.jsonPathKeys.map((path: string) =>
+        path.replaceAll(oldWidgetName, newWidgetName),
+      );
+      return query;
+    });
+}
+
+export function* addNewlyAddedActionsToRedux(actions: Action[]) {
```
Comment (rahulbarwal) [inline app/client/src/sagas/BuildingBlockSagas/index.ts:88]: lets add comments here explaining whats happening.

File: app/client/src/sagas/BuildingBlockSagas/tests/fixtures.ts [hunk 2]
```diff
@@ -101,3 +101,53 @@ export const addEntityAction: ReduxAction<WidgetAddChild> = {
     tabId: "0",
   },
 };
+
+export const newlyCreatedActions: Action[] = [
```
Comment (rahulbarwal) [inline app/client/src/sagas/BuildingBlockSagas/tests/fixtures.ts:105]: Why do we have 2 action dumps, cant we use one of these?
[Other one](https://github.com/appsmithorg/appsmith/pull/34248/files#diff-2eccba5d5605869d8421402908cd31629867854653ab4a23b91f3a5eea36d5f7R30)

**Review by rahulbarwal — CHANGES_REQUESTED** [review]
Requesting minor changes.

File: app/client/src/sagas/BuildingBlockSagas/tests/fixtures.ts (hunk 2 above)
Comment (jacquesikot) [reply app/client/src/sagas/BuildingBlockSagas/tests/fixtures.ts:105]: I have removed the other one in PasteBuildingBlockWidgeSaga and implemented this one from the fixtures file for all.

**Review by rahulbarwal — APPROVED** [review]

**Pull request merged**